### PR TITLE
[NR-89226] Add supportability metric when size > 1MB for each endpoint

### PR DESCRIPTION
**Pull request by ywang-nr**
Ticket: https://new-relic.atlassian.net/browse/NR-89226

**What's changed:**
1. Add supportability metric when payload size > 1MB based on agent spec [here](https://source.datanerd.us/agents/agent-specs/pull/585)

**Testing:**
1. Real device testing

### CODE REVIEW

File: agent-core/src/main/java/com/newrelic/agent/android/agentdata/AgentDataReporter.java [hunk 1]
```diff
@@ -121,6 +122,19 @@ protected void reportCachedAgentData() {
...
     public Future reportAgentData(Payload payload) {
         PayloadSender payloadSender = new AgentDataSender(payload, getAgentConfiguration());
+
+        if (payload.getBytes().length > Constants.Network.MAXPAYLOADSIZELIMIT) {
+            DeviceInformation deviceInformation = Agent.getDeviceInformation();
+            String name = MetricNames.SUPPORTABILITY_MAXPAYLOADSIZELIMIT_ENDPOINT
+                    .replace(MetricNames.TAG_FRAMEWORK, deviceInformation.getApplicationFramework().name())
+                    .replace(MetricNames.TAG_DESTINATION, MetricNames.METRIC_DATA_USAGE_COLLECTOR)
+                    .replace(MetricNames.TAG_SUBDESTINATION, "f");
+            StatsEngine.get().inc(name);
```
Comment (cthomas-newrelic) [inline agent-core/src/main/java/com/newrelic/agent/android/agentdata/AgentDataReporter.java:132]: When recording supportability metrics, use the [SUPPORTABILITY](https://github.com/newrelic/newrelic-android-agent/blob/develop/agent-core/src/main/java/com/newrelic/agent/android/stats/StatsEngine.java#L24) instance or [notice()](https://github.com/newrelic/newrelic-android-agent/blob/develop/agent-core/src/main/java/com/newrelic/agent/android/stats/StatsEngine.java#L38) method. This is an subclass that decorates the metric with some of the fields you referenced here.

Comment (ywang-nr) [reply agent-core/src/main/java/com/newrelic/agent/android/agentdata/AgentDataReporter.java]: Fixed

File: agent-core/src/main/java/com/newrelic/agent/android/harvest/HarvestConnection.java [hunk 2]
```diff
@@ -133,16 +133,29 @@ public HarvestResponse send(HttpURLConnection connection, String message) {
 
             //add supportability metrics
             DeviceInformation deviceInformation = Agent.getDeviceInformation();
-            String name = MetricNames.SUPPORTABILITY_SUBDESTINATION_OUTPUT_BYTES
+            String outputByptesName = MetricNames.SUPPORTABILITY_SUBDESTINATION_OUTPUT_BYTES
```
Comment (cthomas-newrelic) [inline agent-core/src/main/java/com/newrelic/agent/android/harvest/HarvestConnection.java:136]: typo: outputByptesName

File: agent-core/src/main/java/com/newrelic/agent/android/util/Constants.java [hunk 3]
```diff
@@ -34,6 +34,7 @@ public final class Network {
         public static final String CONTENT_LENGTH_HEADER = "Content-Length";
         public static final String USER_AGENT_HEADER = "User-Agent";
         public static final String HOST_HEADER = "Host";
+        public static final long MAXPAYLOADSIZELIMIT = 1000000; //bytes
```
Comment (cthomas-newrelic) [inline agent-core/src/main/java/com/newrelic/agent/android/util/Constants.java:37]: Use case: MAX_PAYLOAD_SIZE

**Review by cthomas-newrelic — DISMISSED** [review]
Minor comments, can be addressed later.

**Review by cthomas-newrelic — APPROVED** [review]
🐑 it

**Pull request merged**